huaweicloud/services/acceptance/dds/resource_huaweicloud_dds_instance_v3_test.go

```diff
@@ -1,32 +1,66 @@
-package huaweicloud
+package dds
 
 import (
 	"fmt"
+	"log"
 	"testing"
 
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
 
 	"github.com/chnsz/golangsdk/openstack/dds/v3/instances"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
 )
 
+func getDdsResourceFunc(c *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := c.DdsV3Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmtp.Errorf("Error creating HuaweiCloud DDS client: %s ", err)
+	}
+
+	opts := instances.ListInstanceOpts{
+		Id: state.Primary.ID,
+	}
+	allPages, err := instances.List(client, &opts).AllPages()
+	if err != nil {
+		return nil, err
+	}
+	instances, err := instances.ExtractInstances(allPages)
+	if err != nil {
+		return nil, err
+	}
+	if instances.TotalCount == 0 {
+		return nil, fmtp.Errorf("dds instance not found.")
+	}
+
+	insts := instances.Instances
+	found := insts[0]
+	return &found, nil
+
+}
+
 func TestAccDDSV3Instance_basic(t *testing.T) {
 	var instance instances.InstanceResponse
-	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	rName := acceptance.RandomAccResourceName()
```

Comment:
`WithDash()`

Reply:
don't need, RandomAccResourceName() is fine